TOML/TOML.sublime-syntax

```diff
@@ -0,0 +1,405 @@
+%YAML 1.2
+---
+# https://toml.io/en/
+# http://www.sublimetext.com/docs/syntax.html
+name: TOML
+scope: source.toml
+version: 2
+
+file_extensions:
+  - toml
+  - tml
+  - Cargo.lock
+  - Gopkg.lock
+  - Pipfile
+  - pdm.lock
+  - poetry.lock
+  - uv.lock
+
+contexts:
+  main:
+    - include: comments
+    - include: sections
+    - include: keys
+    - include: values
+
+###[ COMMENTS ]################################################################
+
+  comments:
+    # https://toml.io/en/v1.0.0#comment
+    - match: \#
+      scope: punctuation.definition.comment.toml
+      push: comment-body
+
+  comment-body:
+    - meta_scope: comment.line.number-sign.toml
+    - match: $\n?
+      pop: 1
+
+###[ SECTIONS ]################################################################
+
+  sections:
+    # https://toml.io/en/v1.0.0#array-of-tables
+    - match: \[\[
+      scope: meta.section.toml punctuation.definition.section.begin.toml
+      push: array-section-name-body
+    # https://toml.io/en/v1.0.0#table
+    - match: \[
+      scope: meta.section.toml punctuation.definition.section.begin.toml
+      push: single-section-name-body
+
+  array-section-name-body:
+    - meta_content_scope: meta.section.toml entity.name.section.toml
+    - match: \]\]
+      scope: meta.section.toml punctuation.definition.section.end.toml
+      set: expect-eol
+    - include: name-content
+    - include: illegal-eol-pop
+
+  single-section-name-body:
+    - meta_content_scope: meta.section.toml entity.name.section.toml
+    - match: \]
+      scope: meta.section.toml punctuation.definition.section.end.toml
+      set: expect-eol
+    - include: name-content
```

Comment:
This include applies the same rules to the section name as to individual keys. While I understand it is done for consistency/symmetry, I want to argue that section headers should be treated different from individual keys in a way that makes `entity.name.section` stand out more. 

Since I'm also arguing for using unquoted string scopes for key segments, these would make `string` selectors more specific and thus make them take precedence over an `entity.name` selector. Personally, I assign a background color to this scope, but most people do not and I wouldn't want to break their section highlighting into a string. 

Thus, my proposal is to only apply `string` scopes within individual keys and `meta.string` within section names.

Reply:
Section names and keys have identical rules in TOML. Thus I struggle with applying different rules on them, so much.